File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1] * len(nums)
        """
        [1,1,1,1]
             ^ 
        [1,2,4,6]
             ^
        [1,1,2,8]
        [48,24,12,8]
        """

        for i in range(1, len(nums)):
            res[i] = res[i-1] * nums[i-1]
        post = 1
        for i in range(len(nums) - 1, -1, -1):
            res[i] *= post
            post *= nums[i]
        return res
            

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> num_set(nums.begin(), nums.end());
        int res = 0; 

        for (int n: num_set)
        {
            if (num_set.find(n - 1) == num_set.end())
            {
                int longest = 1;
                while (num_set.find(n+1) != num_set.end())
                {
                    longest++;
                    n++;
                }
                res = max(res, longest);
            }
        }
        return res;
    }
};
